spec/controllers/groceries_controller_spec.rb

```diff
@@ -32,22 +41,37 @@
   end
 
   describe 'POST create' do
-    grocery_params = { name: "Test", description: "Test" }
+    let(:grocery_params) {
+       {
+         name: 'Test',
+         description: 'Test'
+       }
+    }
     subject { post :create, grocery: grocery_params, user_group_id: user_group }
 
-    it 'should set the default group if nil' do
-      expect(controller.current_user.default_group).to be_nil
-      subject
-      expect(controller.current_user.default_group).to eq user_group
+    context 'grocery is valid' do
+      it 'should create a new grocery' do
+        expect { subject }.to change(Grocery, :count).by 1
+      end
+
+      it 'should set the default group if nil' do
+        expect(controller.current_user.default_group).to be_nil
+        subject
+        expect(controller.current_user.default_group).to eq user_group
+      end
+    end
+
+    context 'grocery is invalid' do
+      it 'should redirect to new template' do
+        grocery_params[:name] = ""
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.